app/controllers/vendor_api/test_data_controller.rb

```diff
@@ -18,6 +18,7 @@ def generate
         provider: current_provider,
         courses_per_application: courses_per_application_param,
         count: count_param,
+        previous_cycle: previous_cycle_param,
```

Comment:
Why not allow any cycle year here?

Reply:
Not sure I follow, the ticket is about supporting generating applications in previous recruitment cycle

Reply:
We pass a boolean to indicate previous / current cycle, which limits our options internally when calling `GenerateTestApplicationsForProvider`. We could have the controller only permit previous or current cycle year as a param value but allow `GenerateTestApplicationsForProvider` to generate for any recruitment cycle year without any real additional complexity. Admittedly it's future-proofing for a need we don't yet have but it seems more flexible to allow any cycle. Not a blocker, just a suggestion.